parcels/field.py

```diff
@@ -1193,7 +1193,7 @@ def read_lonlat(self):
         if len(lon.shape) > 1:  # if lon, lat are rectilinear but were stored in arrays
             xdim = lon_subset.shape[0]
             ydim = lat_subset.shape[1]
-            if np.allclose(lon_subset[0, :], lon_subset[int(xdim/2), :]) and np.allclose(lat_subset[:, 0], lat_subset[:, int(ydim/2)]):
+            if np.allclose(lon_subset[0, :], lon_subset[xdim-1, :]) and np.allclose(lat_subset[:, 0], lat_subset[:, ydim-1]):
```

Comment:
The point of this check was to quickly assess if a mesh was curvilinear.
If that test is not sufficient, we should maybe have a more complete one checking all the rows and lines, or one out of XX rows and lines, but checking 1 and last can be problematic, especially if a mesh is periodic.

Reply:
OK, well maybe we need to check all rows and columns then, because the mi-way rows and columns failed for MOM.
But perhaps we should even aks ourselves why we need this test. Even if a grid is RectiLinear, treating it as Curvilinear here would not slow down the code by much, would it? Shouldn't we just assume that any 2d array of Longitudes or latitudes always signals a CurviLinear Grid?

Reply:
I don't really know here about timing. We could quantify that.

Reply:
I've now written a code that checks for all lons and lats, but does it quite efficient to stop as soon at it finds the first lon that is different from the first column. Since we don't expect to see 2D Rectilinear Grids often (in which case we have to search all lons and lats), I think this is a good solution. Agree?